### DISCUSSION TITLE
Grid view for a single DAG run

### DISCUSSION THREAD

**Discussion by jayesh-shaw-glean** [apache/airflow#29571] [Q&A]
### Description

A filter on run_id in the grid view

### Use case/motivation

Having used Airflow for sometime, I and my teammates get the most value out of the grid view. I might be missing something, but it seems that there is no way to view a single dag run in the grid view, if it isn't the latest run (latest success/running/failed runs also work). I would like to know if there is some way to get to view a single dag run or if there isn't then it would be a great feature to add. 

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by Taragolis** [comment]
Could you a bit clarify, what do you mean "it seems that there is no way to view a single dag run in the grid view"

Is that what are you looking for? Or something else?

https://user-images.githubusercontent.com/3998685/219343273-faeb27e6-e648-404f-971e-b64ace6622ed.mp4

**Reply by jayesh-shaw-glean** [reply]
> Could you a bit clarify, what do you mean "it seems that there is no way to view a single dag run in the grid view"

Apologies for not being clear in the post itself! I meant in the grid view is it possible to show a single DAG run and not more than one. Somewhat like this:

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/109058521/219944241-490a6e60-8a60-4960-aef6-673663e2fabe.png">

As you can see even though there might be many runs of a DAG, we can view just one of them and get its details from this view. Currently we only have `num_runs` and `status` query parameters for this view, and setting `num_runs=1` only shows the last dag run with the given `status` (if any, otherwise over all dag runs). 

The use case is that we have people who would need to interact with airflow but aren't too familiar with it. To be able to point them to the grid view of a particular dag run would be very useful for us.

**Reply by Taragolis** [reply]
Grid view it is successor of Tree view (since Airflow 2.3), which always show multiple DAG Runs.
Grid view not only show this DAG Runs but also allow iterate with it, by select specific DAG, show some statistics.

You still could select specific DAG Run from subsets of DAG Runs and get details

![image](https://user-images.githubusercontent.com/3998685/219954358-9a4e6c25-3f45-4e09-b68c-539326791179.png)

So maybe that the reason why `num_runs` work so specifically if set to `1`

**Reply by jayesh-shaw-glean** [reply]
Yeah, the issue is that even if we can point to a single DAG run in a grid view and get its details easily, it is quite easy to mis click on another DAG run when you wanted to get info about a specific DAG run 😅
